src/components/gauge/gauge.tsx

```diff
@@ -0,0 +1,381 @@
+
+  /**
+   *  Allows display current value. Default is disabled.
+   *
+   */
+  @Prop() showValue = false;
+
+  /**
+   * The Minimum Value of the gauge
+   *
+   */
+  @Prop() minValue: number;
+
+  /**
+   * The Currecnt value indicanding in the gauge
+   *
+   */
+  @Prop() currentValue: number = this.minValue;
+
+  /**
+   * This allows specifying the width of the circumference _(When gauge is Circle type)_ and the width of the bar _(When gauge is Line type)_ in % relative the component size.
+   *
+   */
+  @Prop() thickness = 10;
+
+  @State() maxValue: number;
+
+  @State() totValues = this.minValue;
+
+  @State() children = [];
+
+  @State() minorSize: number;
+
+  @Listen("gxGaugeRangeDidLoad")
+  onGaugeRangeDidLoad({ detail: childRange }) {
+    this.children = [...this.children, childRange];
+    this.totValues += childRange.amount;
+    childRange.element.addEventListener("gxGaugeRangeDidUnload", () => {
+      const index = this.children.findIndex(x => x === childRange);
+      this.children.splice(index, 1);
+      this.totValues -= childRange.amount;
+    });
+    childRange.element.addEventListener("gxGaugeRangeDidUpdate", () => {
+      const index = this.children.findIndex(x => x === childRange);
+      this.children.splice(index, 1, childRange);
+      this.totValues = 0;
+      for (const childInstance of this.children) {
+        this.totValues += childInstance.amount;
+      }
+    });
+  }
+
+  componentWillLoad() {
+    this.totValues = 0;
+  }
+
+  private calcThickness() {
+    return typeof this.thickness === "number" &&
+      (this.thickness > 0 && this.thickness <= 100)
+      ? this.thickness / 5
+      : 10;
+  }
+
+  private calcPercentage() {
+    return (
+      (this.currentValue - this.minValue) *
+      100 /
+      (this.maxValue - this.minValue)
+    );
+  }
+
+  private renderCircle(childRanges) {
+    const svgRanges = [];
+    let acumulation = 0;
+    //////////////////////////////////////////
+    function calcPositionRange(preValue, value) {
+      acumulation += preValue;
+      return value + acumulation;
+    }
+
+    function addSVGCircle(currentChild, nextChild, component) {
+      return (
+        <circle
+          r="39.59%"
+          cx="50%"
+          cy="50%"
+          stroke={currentChild.color}
+          stroke-dasharray={`${2.488 /
+            (component.totValues / 100) *
+            calcPositionRange(
+              !!nextChild ? parseInt(nextChild.getAttribute("amount"), 10) : 0,
+              parseInt(currentChild.getAttribute("amount"), 10)
+            )}, 248.16`}
+          fill="none"
+          amount={currentChild.amount}
+          stroke-width={`${component.calcThickness()}%`}
+        />
+      );
+    }
+    //////////////////////////////////////////
+    for (let i = childRanges.length - 1; i >= 0; i--) {
+      svgRanges.push(addSVGCircle(childRanges[i], childRanges[i + 1], this));
+    }
+    svgRanges.reverse();
```

Comment:
Maybe instead of reversing this array, we should just add the items to the beginning of the array?

Reply:
I made it in that way for take the order of HTML and draw the ranges.
It's like... _"The last range added is the first range in the gauge"_
![image](https://user-images.githubusercontent.com/33690842/60980574-18f22180-a30b-11e9-8671-18ae6e820f92.png)

Reply:
I'm just suggesting an alternative way of having the items added to the array so you don't have to reverse it.

Reply:
I was looking for another way to draw each `<circle>` of SVG. But I need to first reverse the list of  `<gx-gauge-range>`  that are within  `<gx-gauge>`, then passing one by one and its next for calculate the `stroke-dasharray`. And then reverse the list of `<circle>`s to add all `<circle>`s  within the SVG (with the `stroke-dasharray` already calculated)

Reply:
Changes done! commit 3c4d33abc5cf0d9f0a62ddbde15cbf38f1846857